traitcuration/traits/datasources/clinvar.py

```diff
@@ -0,0 +1,77 @@
+import urllib.request
+import csv
+import gzip
+import shutil
+import itertools
+import os
+from ..models import Trait
+
+
+def extract_clinvar_data():
+    """
+    This function downloads the latest ClinVar TSV release data and extracts it into a 
+    'variant_summary.txt' file
+    """
+    print("Downloading ClinVar data...")
+    url = 'https://ftp.ncbi.nlm.nih.gov/pub/clinvar/tab_delimited/variant_summary.txt.gz'
+    urllib.request.urlretrieve(url, './variant_summary.txt.gz')
+    print("Extracting ClinVar data...")
+    with gzip.open('variant_summary.txt.gz', 'rb') as f_in:
+        with open('variant_summary.txt', 'wb') as f_out:
+            shutil.copyfileobj(f_in, f_out)
+
+
+def parse_trait_names_and_source_records():
+    """
+    This function parses a downloaded 'variant_summary.txt' file, and returns a unique set of traint names
+    along with their calculated source record number, in a form of a dictionary where the key is the trait name
+    and the value is the source record number
+    """
+    print("Parsing ClinVar data...")
+    with open('variant_summary.txt') as tsvfile:
+        reader = csv.DictReader(tsvfile, dialect='excel-tab')
+        # A dictionary with trait names as keys and sets of source records as values
+        traits_dict = dict()
+        # The int value here defines how many records should be parsed
+        for row in itertools.islice(reader, 10):
+            # For every row, get its allele_id and all its rcv_accessions and phenotypes
+            row_alleleid = [(row['#AlleleID'])]
+            row_rcv_list = set(row['RCVaccession'].split(';'))
+            row_phenotype_list = set(row['PhenotypeList'].split(';'))
+            # Get every possible combination tuple of allele_id rcv_accessions and phenotypes for the current row
+            tuple_list = list(itertools.product(
```

Comment:
I think `list(...)` wrap is redundant here. itertools.product returns a generator which can and should be iterated directly.

Reply:
Yes, since I had the source record calculation wrong, I removed this line altogether